R/utilities-plots.R

```diff
@@ -350,6 +350,49 @@ prettyCaption <- function(captions, plotObject, element = "legend") {
   return(addLineBreakToCaption(captions, width = maxWidth))
 }
 
+#' @title updateWatermarkDimensions
+#' @description Update Watermark dimensions
+#' @param plotObject A `ggplot` object
+#' @return A `ggplot` object
+#' @keywords internal
+updateWatermarkDimensions <- function(plotObject) {
+  # No need to update if no displayed watermark
+  if (isEmpty(plotObject$plotConfiguration$background$watermark$text)) {
+    return(plotObject)
+  }
+  # Watermark size in inches to compare with plot dimensions
+  # Font size is in point = 1/72 inches
+  watermarkSize <- nchar(plotObject$plotConfiguration$background$watermark$text) *
+    plotObject$plotConfiguration$background$watermark$font$size / 72
+  watermarkWidth <- abs(watermarkSize * cos(plotObject$plotConfiguration$background$watermark$font$angle * pi / 180))
+  watermarkHeight <- abs(watermarkSize * sin(plotObject$plotConfiguration$background$watermark$font$angle * pi / 180))
```

Comment:
Why `*pi/180` ?

Reply:
Because the angle from `font` is in degrees and the `cos`/`sin` functions use radians.
So it needs to be converted to get the correct width/height if there is an angle in the watermark.
For the conversion,  $360^o = 2 \pi$